File: javascript/nth-prime/nth-prime.js
//
// This is only a SKELETON file for the 'Nth Prime' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const prime = (n) => {
  if (n === 0) {
    throw new Error('there is no zeroth prime');
  };
  let [number, primes] = [2, []];
  while (primes.length < n) {
    let isPrime = true;
    for (const prime of primes) {
      if (number % prime === 0) {
        isPrime = false;
        break;
      };
    };
    if (isPrime) {
      primes.push(number);
    };
    number++;
  };
  return primes[primes.length - 1];
};

File: python/nth-prime/nth_prime.py
def prime(number):
    if number == 0:
        raise ValueError("there is no zeroth prime")
    primes, test = [2], 3
    while len(primes) < number:
        is_prime = True
        for prime in primes:
            if test % prime == 0:
                is_prime = False
                break
        if is_prime:
            primes.append(test)
        test += 1
    return primes[-1]
